source/agora/node/FullNode.d

```diff
@@ -1062,6 +1062,17 @@ public class FullNode : API
         return this.enroll_man.getEnrollment(enroll_hash);
     }
 
+    /// GET: /enrollments
+    public override Enrollment[] getEnrollments (ulong height, ulong enroll_count)
+        @safe
+    {
+        auto enrolls = this.enroll_man.enroll_pool.getEnrollments(Height(height));
+        if (enrolls.length > enroll_count)
+            return enrolls;
+        
+        return Enrollment[].init;
+    }
+
```

Comment:
I don't understand this API. I don't see why it would make sense from the point of view of either the client, or the server.

In the client:
- You probably want to set this to a low value (I would just use 0?) because otherwise you might not get any result;

In the server:
- You always allocate, then check the length, then discard the allocated array if the user-provided value is too high;
- If an attacker repeatedly sends you `ulong.max`, you'll just end up allocating like crazy;

So the endpoint is an easy DoS vector, and doesn't make sense to the client either. What am I missing ?